small/lsregion.c

```diff
@@ -91,3 +92,46 @@ lsregion_aligned_reserve_slow(struct lsregion *lsregion, size_t size,
 	assert((char *)pos + size <= (char *)lslab_end(slab));
 	return pos;
 }
+
+void
+lsregion_to_iovec(const struct lsregion *lsregion, struct iovec *iov,
+		  int *iovcnt, struct lsregion_svp *svp)
+{
+	struct lslab *lslab;
+	int64_t prev_id = svp->max_id;
+	size_t prev_pos = svp->pos;
+	int max_iovcnt = *iovcnt;
+	int cnt = 0;
+	rlist_foreach_entry(lslab, &lsregion->slabs.slabs, next_in_list) {
+		/* An unused slab. */
+		if (lslab->max_id == LSLAB_NOT_USED_ID)
+			continue;
+		/* An already seen slab with no new allocations. */
+		if (lslab->max_id <= prev_id)
+			continue;
+		if (cnt >= max_iovcnt)
+			break;
+		if (lslab->max_id >= svp->max_id) {
```

Comment:
Shouldn't this condition always be true here?

Reply:
If we tell the user to perform `lsregion_gc` after each `lsregion_to_iovec` it will always be true, indeed. Otherwise we have to skip all the flushed but not yet gc-ed slabs.

It's tempting to make `lsregion_to_iovec` automatically `gc` everything that was flushed, but I can't do this: i only save the slab pointers, not their contents, so the slabs have to live until the user dumps iovec contents wherever he wants.